### PR TITLE
[FIX] OD-1476: exclude all picking lines when freeing reservations

### PR DESCRIPTION
**Pull request by hbrunn**
I propose to add this to our feature branch and then bump the hash there.

Shouldn't we PR this whole branch into OCA too?

Buildout PR: https://github.com/VanMoof/odoo-buildout/pull/296

### CODE REVIEW

File: stock_move_manual_lot/models/stock_move_line.py
Comment (hbrunn) [inline stock_move_manual_lot/models/stock_move_line.py]: we need this because this function isn't called on all move lines at once, and we actually don't want this function to mess with lines from our own picking(s) anyways.

The problematic line is https://github.com/OCA/OCB/blob/12.0/addons/stock/models/stock_move_line.py#L526

**Review by astirpe — APPROVED** [review]
Works fine now! :+1:

**Review by StefanRijnhart — CHANGES_REQUESTED** [review]
~~This PR does not add a test to reproduce the problem it tries to solve, and it introduces a regression in the case multiple move lines of the same picking are updated, breaking test 03.~~ Comments addressed :sun_with_face:

File: stock_move_manual_lot/models/stock_picking.py
Comment (StefanRijnhart) [inline stock_move_manual_lot/models/stock_picking.py]: Here too, the fact that the previous version worked is witnessed by the tests that were added at that time. If this does fix a problem, please add a test to reproduce it.

**Comment by hbrunn** [comment]
now with a different approach and tests

**Comment by StefanRijnhart** [comment]
Looking good! As it happens, I was working on the same issue and I'm proposing a complementary part of the solution into this branch here: https://github.com/VanMoof/stock-logistics-workflow/pull/15

**Comment by hbrunn** [comment]
...and another rewrite. please re-review

**Comment by StefanRijnhart** [comment]
Excellent improvement.

Waiting on your review of my PR into this one before my approval.

**Comment by hbrunn** [comment]
thanks, now we're there I think

**Review by StefanRijnhart — APPROVED** [review]
I've restarted tests on the buildout branch as the final validation

**Pull request merged**
